src/test/java/formflow/library/submissions/actions/FormatDateBeforeSave.java

```diff
@@ -0,0 +1,17 @@
+package formflow.library.submissions.actions;
+
+import formflow.library.config.submission.Action;
+import formflow.library.data.Submission;
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("unused")
+public class FormatDateBeforeSave implements Action {
+  public void run(Submission submission) {
+    List<String> dateComponents = new ArrayList<>(3);
+    dateComponents.add((String) submission.getInputData().get("dateMonth"));
+    dateComponents.add((String) submission.getInputData().get("dateDay"));
+    dateComponents.add((String) submission.getInputData().get("dateYear"));
+    submission.getInputData().put("formattedDate", String.join("/", dateComponents));
```

Comment:
:) When I left my comment above about where to run the save from, I thought of this scenario!